src/models/budget/impls.rs

```diff
@@ -193,31 +170,65 @@ impl TaskBudgets {
             let time_slots_iterator =
                 TimeSlotsIterator::new(start, deadline, goal.repeat, goal.filters.clone());
 
-            for time_slots in time_slots_iterator {
+            for timeline in time_slots_iterator {
                 let task_id = *counter;
                 *counter += 1;
-                if !time_slots.is_empty() {
-                    let task = Task {
-                        id: task_id,
-                        goal_id: goal.id.clone(),
+                if !timeline.slots.is_empty() {
```

Comment:
```suggestion
                if timeline.slots.len() > 0 {
```
Clean code standard to not use (double) negatives in 'if' statements.
It is clearer when you say: only use postitives in 'if' statements. Any negative can be converted to a posititve.

Reply:
I think I will return it back as below warning by `cargo clippy`:
```rust
❯ cargo clippy
warning: length comparison to zero
   --> src/models/budget/impls.rs:176:20
    |
176 |                 if timeline.slots.len() > 0 {
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^ help: using `!is_empty` is clearer and more explicit: `!timeline.slots.is_empty()`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_zero
    = note: `#[warn(clippy::len_zero)]` on by default
```